Packs/Anomali_ThreatStream/Integrations/Anomali_ThreatStream_v2/Anomali_ThreatStream_v2.py

```diff
@@ -87,37 +77,60 @@
-                       'threatstream-get-analysis-status', 'threatstream-analysis-report']:
-            # in order to prevent raising en error in case model/indicator/report was not found
-            return {}
-        else:
-            return_error("The resource not found. Check the endpoint.")
-    elif res.status_code not in {200, 201, 202}:
-        return_error(F"Error in API call to ThreatStream {res.status_code} - {res.text}")
-
-    if text_response:
-        return res.text
-    return res.json()
+class Client:
+    def __init__(self, base_url, use_ssl, default_threshold, reliability):
+        self.base_url = base_url
+        self.use_ssl = use_ssl
+        self.default_threshold = default_threshold
+        self.reliability = reliability
+
+    def http_request(self, method, url_suffix, params=None, data=None, headers=None, files=None, json=None, text_response=None):
+        """
+            A wrapper for requests lib to send our requests and handle requests and responses better.
+        """
+        res = requests.request(
+            method,
+            self.base_url + url_suffix,
+            verify=self.use_ssl,
+            params=params,
+            data=data,
+            headers=headers,
+            files=files,
+            json=json
+        )
+        # Handle error responses gracefully
+        if res.status_code in {401}:
+            return_error("Got unauthorized from the server. Check the credentials.")
+        elif res.status_code in {404}:
+            command = demisto.command()
+            if command in ['threatstream-get-model-description', 'threatstream-get-indicators-by-model',
+                           'threatstream-get-analysis-status', 'threatstream-analysis-report']:
+                # in order to prevent raising en error in case model/indicator/report was not found
+                return {}
+            else:
+                return_error("The resource not found. Check the endpoint.")
+        elif res.status_code not in {200, 201, 202}:
+            return_error(F"Error in API call to ThreatStream {res.status_code} - {res.text}")
+
+        if text_response:
+            return res.text
+        return res.json()
+
+    def get_dbot_context(self, indicator, threshold, mapping=None):
+        """
+             Builds and returns dictionary with Indicator, Type, Vendor and Score keys
+             and values from the indicator that will be returned to context.
+        """
+        if not mapping:
+            mapping = DBOT_MAPPING
+        dbot_context = {mapping[k]: v for (k, v) in indicator.items() if k in mapping.keys()}
+        indicator_score = DBOT_SCORE[indicator.get('meta', {}).get('severity', 'low')]
+        # the indicator will be considered as malicious in case it's score is greater or equal to threshold
+        dbot_context['Score'] = 3 if indicator_score >= DBOT_SCORE[threshold] else indicator_score
+        dbot_context['Vendor'] = 'ThreatStream'
```

Comment:
I suggest putting `ThreatStream` in a global var, so you can use it in here, error messages and in the contextpaths you create along the integration commands.

Reply:
done.